### PR TITLE
Use a lru cache for MainAccountStateDB._get_account()

### PR DESCRIPTION
**Pull request by gsalgado**
Part of the fixes for #542

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
         "rlp>=0.4.7,<1.0.0",
         "eth-keys>=0.2.0b3,<1.0.0",
         "trie>=1.3.2,<2.0.0",
+        "pylru>=1.0.9",
```
Comment (pipermerriam) [inline setup.py]: It looks like https://pypi.python.org/pypi/lru-dict might be worth evaluating as it's implemented in C.

File: evm/db/state.py [hunk 2]
```diff
@@ -294,7 +316,13 @@ def account_is_empty(self, address):
     # Internal
     #
     def _get_account(self, address):
-        rlp_account = self._trie[address]
+        cache_key = (self._unwrapped_db, self.root_hash, address)
+        if cache_key in account_cache:
+            rlp_account = account_cache[cache_key]
+        else:
+            rlp_account = self._trie[address]
+            account_cache[cache_key] = rlp_account
```
Comment (pipermerriam) [inline evm/db/state.py]: I've been a fan of the following pattern for cached functions.  I find it easier to parse.

```python
def cached_fn():
    if key not in cache:
        cache[key] = get_value()

    value = cache[key]
```

It's a minor improvement but it's a smaller conditional block to parse and the *value* always comes from the same place.

File: setup.py (hunk 1 above)
Comment (gsalgado) [reply setup.py]: For this limited one-block-import test I'm running it doesn't seem to provide any benefits. I guess it depends how large this cache can get. Maybe just leave a comment there to evaluate it properly if `_get_account()` shows up again as one of the worst offenders in cProfile stats?

File: evm/db/state.py (hunk 2 above)
Comment (gsalgado) [reply evm/db/state.py]: I like that!

File: setup.py (hunk 1 above)
Comment (carver) [reply setup.py]: :-1: `pylru` is GPL https://github.com/jlhutch/pylru/blob/master/LICENSE.txt

Comment (gsalgado) [reply setup.py]: Should we go with lru-dict, then?

Comment (pipermerriam) [reply setup.py]: @gsalgado I think yes, (that's what web3.py is using at least).

**Pull request merged**